sys/net/link_layer/ble_ll/ble_ll.c

```diff
@@ -0,0 +1,331 @@
+                pkt->next->data, ble_pkt_size);
+    }
+
+    ble_pkt = ng_pktbuf_add(NULL, payload, ble_pkt_size, NG_NETTYPE_UNDEF);
+    ng_pktbuf_release(pkt);
+
+    return dev->driver->send_data(dev, ble_pkt);
+}
+
+/**
+ * @brief    Handle incoming link layer packet
+ */
+static void _receive(ng_pktsnip_t *pkt)
+{
+    ble_pkt_t *data;
+
+    data = (ble_pkt_t *)pkt->data;
+
+    if (_ll_state == LL_ADVERTISING) {
+        /* Send respone to received BLE SCAN_REQ packet */
+        if (data->header.pdu_type == SCAN_REQ_TYPE) {
+            DEBUG("ble_ll: receive_data -> SCAN_REQ received\n");
+            /*_send_scanresponse_pkt();*/
+        }
+
+        /* Send respone to received BLE CONNECT_REQ packet */
+        else if (data->header.pdu_type == CONNECT_REQ_TYPE) {
+            DEBUG("ble_ll: receive_data -> CONNECT_REQ received\n");
+        }
+    }
+
+#if ENABLE_BLE_PKT_DEBUG
+    int i;
+    static const char *pdu_type_name[] = {
+        "ADV_IND", "ADV_DIRECT_IND", "ADV_NONCONN_IND", "SCAN_REQ", "SCAN_RSP",
+        "CONNECT_REQ", "ADV_SCAN_IND"
+    };
+
+    printf("Type: %s  Len: %u %s Address: %02x:%02x:%02x:%02x:%02x:%02x\n",
+           pdu_type_name[data->header.pdu_type], data->header.length,
+           (data->header.pdu_type == SCAN_REQ_TYPE) ? "Scanning" : "Advertising",
+           data->payload[5], data->payload[4], data->payload[3], data->payload[2], data->payload[1],
+           data->payload[0]);
+
+    for (i = 0; i < data->header.length; i++) {
+        printf("%02x ", data->payload[i]);
+    }
+
+    printf("\n");
+#endif
+
+}
+
+
+/**
+ * @brief   Function called by the device driver on device events
+ *
+ * @param[in] event         type of event
+ * @param[in] data          optional parameter
+ */
+static void _event_cb(ng_netdev_event_t event, void *data)
+{
+    DEBUG("ble_ll: event triggered -> RX_COMPLETE\n");
+    /* NOMAC only understands the RX_COMPLETE event... */
+    if (event == NETDEV_EVENT_RX_COMPLETE) {
+        ng_pktsnip_t *pkt;
+        ng_netreg_entry_t *sendto;
+
+        /* get pointer to the received packet */
+        pkt = (ng_pktsnip_t *)data;
+        /* handle recieved packet */
+        _receive(pkt);
+        /* find out, who to send the packet to */
+        sendto = ng_netreg_lookup(pkt->type, NG_NETREG_DEMUX_CTX_ALL);
+        /* throw away packet if no one is interested */
+        if (sendto == NULL) {
+            DEBUG("ble_ll: unable to forward packet of type %i (sendto == NULL)\n", pkt->type);
+            ng_pktbuf_release(pkt);
+            return;
+        }
+        /* send the packet to everyone interested in it's type */
+        ng_pktbuf_hold(pkt, ng_netreg_num(pkt->type, NG_NETREG_DEMUX_CTX_ALL) - 1);
+        while (sendto != NULL) {
+            DEBUG("ble_ll: sending pkt %p to PID %u\n", (void*)pkt, sendto->pid);
+            ng_netapi_receive(sendto->pid, pkt);
+            sendto = ng_netreg_getnext(sendto);
+        }
+    } else {
```

Comment:
According to our [coding conventions](https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions), please put `else` in a new line